routers/api/v1/repo/issue.go

```diff
@@ -462,6 +462,24 @@ func ListIssues(ctx *context.APIContext) {
 		isPull = util.OptionalBoolNone
 	}
 
+	if isPull != util.OptionalBoolNone && !ctx.Repo.CanWriteIssuesOrPulls(isPull.IsTrue()) {
```

Comment:
Why it checks `CanWriteIssuesOrPulls` for `ListIssues` handler?

Reply:
Because this function could list both issues and pull requests. One maybe have only issue or pull request read permission. So if he has only issue read permission, but he inputs a parameter clearly to request pull request, then we should return no permission.

Reply:
Why `CanWriteIssuesOrPulls` but not `CanReadIssuesOrPulls`?

Reply:
typo